server/index.js

```diff
@@ -30,10 +34,38 @@ app.get('/api/products', (req, res, next) => {
       res.status(200).json(products);
     })
     .catch(err => {
-      console.error(err);
-      res.status(500).json({
-        error: 'An unexpected error occurred.'
-      });
+      next(err);
+    });
+});
+
+app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+  if (!parseInt(productId, 10)) {
+    next(new ClientError('"productId" must be a positive integer', 400));
+  }
```

Comment:
Be sure to include return statements so that your server doesn't attempt to respond twice